### ISSUE TITLE
Bug: After link input the formatting is lost

### ISSUE THREAD

**Issue by 5tausend** [facebook/lexical#5089]
Hello everyone,

there is a formatting bug that happens when a link is added. This bug already exists in the Lexical-playground.

![image](https://github.com/facebook/lexical/assets/42815348/4926c6b4-32ed-4b61-9f6b-6d86b4515f75)

![image](https://github.com/facebook/lexical/assets/42815348/97473f82-ce82-4600-af94-0b8da62a522c)

does anyone have a solution?

**Comment by tahir606** [comment]
Hey! I have been able to replicate this. And working on a solution! Cheers!

**Comment by tahir606** [comment]
Hi guys, I have modified the setting element format structure in this PR. Basically, when a link is formatted the format is added to the parent paragraph node instead of the link node. This is the generic functionality in all of the format setting. 

So I have added a check for the link node, and if it is a link node a matching parent is picked and the format is applied. This fixes the issue that is being displayed above.

Would love a review on this.

Regards
Tahir

https://github.com/facebook/lexical/assets/20771570/bfaf7122-ca42-4b06-aa69-5720db238d03
